File: app/src/main/java/com/cooper/wheellog/utils/BaseAdapter.java
package com.cooper.wheellog.utils;

import android.content.Context;

public abstract class BaseAdapter {
    protected Context mContext;

    public BaseAdapter setContext(Context context) {
        mContext = context;
        return this;
    }

    public abstract boolean decode(byte[] data);

    public void updatePedalsMode(int pedalsMode) {
    }

    public void setLightMode(int lightMode) {
    }

    public void setRollAngleMode(int rollAngleMode) {
    }

    public void updateBeeperVolume(int beeperVolume) {
    }

    public void setMilesMode(boolean milesMode) {
    }

    public void setLightState(boolean on) {
    }

    public void setLedState(boolean on) {
    }

    public void setTailLightState(boolean on) {
    }

    public void setHandleButtonState(boolean on) {
    }

    public void setBrakeAssist(boolean on) {
    }

    public void setLedColor(int value, int ledNum) {
    }

    public void setAlarmEnabled(boolean on, int num) {
    }

    public String getLedModeString() {
        return "";
    }

    public Boolean getLedIsAvailable(int ledNum) {
        return false;
    }

    public void setLimitedModeEnabled(boolean on) {
    }

    public void setLimitedSpeed(int value) {
    }


    public void setAlarmSpeed(int value, int num) {
    }

    public void setRideMode(boolean on) {
    }

    public void setLockMode(boolean on) {
    }

    public void setTransportMode(boolean on) {
    }

    public void setDrl(boolean on) {
    }

    public void setGoHomeMode(boolean on) {
    }

    public void setFancierMode(boolean on) {
    }

    public void setMute(boolean on) {
    }

    public void setFanQuiet(boolean on) {
    }

    public void setFan(boolean on) {
    }

    public void setLightBrightness(int value) {
    }

    public void powerOff() {
    }

    public void switchFlashlight() {
    }

    public void wheelBeep() {
    }

    public void updateMaxSpeed(int wheelMaxSpeed) {
    }

    public void setSpeakerVolume(int speakerVolume) {
    }

    public void setPedalTilt(int angle) {
    }

    public void setPedalSensivity(int sensivity) {
    }

    public void wheelCalibration() {
    }

    public void updateLedMode(int ledMode) {
    }

    public void updateStrobeMode(int strobeMode) {
    }

    public void updateAlarmMode(int alarmMode) {
    }

    public int getCellSForWheel() {
        return 0;
    }

    public boolean isReady() {
        return false;
    }
}

File: app/src/main/java/com/cooper/wheellog/utils/BaseAdapter.kt
package com.cooper.wheellog.utils

import android.content.Context
import com.cooper.wheellog.WheelLog

abstract class BaseAdapter {
    val context: Context?
        get() = WheelLog.appContext

    abstract fun decode(data: ByteArray?): Boolean
    open fun updatePedalsMode(pedalsMode: Int) {}
    open fun setLightMode(lightMode: Int) {}
    open fun setRollAngleMode(rollAngleMode: Int) {}
    open fun updateBeeperVolume(beeperVolume: Int) {}
    open fun setMilesMode(milesMode: Boolean) {}
    open fun setLightState(on: Boolean) {}
    open fun setLedState(on: Boolean) {}
    open fun setTailLightState(on: Boolean) {}
    open fun setHandleButtonState(on: Boolean) {}
    open fun setBrakeAssist(on: Boolean) {}
    open fun setLedColor(value: Int, ledNum: Int) {}
    open fun setAlarmEnabled(on: Boolean, num: Int) {}
    open val ledModeString: String?
        get() = ""

    open fun getLedIsAvailable(ledNum: Int): Boolean {
        return false
    }

    open fun setLimitedModeEnabled(on: Boolean) {}
    open fun setLimitedSpeed(value: Int) {}
    open fun setAlarmSpeed(value: Int, num: Int) {}
    open fun setRideMode(on: Boolean) {}
    open fun setLockMode(on: Boolean) {}
    open fun setTransportMode(on: Boolean) {}
    open fun setDrl(on: Boolean) {}
    open fun setGoHomeMode(on: Boolean) {}
    open fun setFancierMode(on: Boolean) {}
    open fun setMute(on: Boolean) {}
    open fun setFanQuiet(on: Boolean) {}
    open fun setFan(on: Boolean) {}
    open fun setLightBrightness(value: Int) {}
    open fun powerOff() {}
    open fun switchFlashlight() {}
    open fun wheelBeep() {}
    open fun updateMaxSpeed(wheelMaxSpeed: Int) {}
    open fun setSpeakerVolume(speakerVolume: Int) {}
    open fun setPedalTilt(angle: Int) {}
    open fun setPedalSensivity(sensivity: Int) {}
    open fun wheelCalibration() {}
    open fun updateLedMode(ledMode: Int) {}
    open fun updateStrobeMode(strobeMode: Int) {}
    open fun updateAlarmMode(alarmMode: Int) {}
    open val cellsForWheel: Int
        get() = 1
    open val isReady: Boolean
        get() = false
}